src/js/core/methods/helpers/cardanoCertificate.js

```diff
@@ -122,38 +125,62 @@ const validatePoolParameters = (poolParameters: CardanoPoolParameters) => {
     }
 };
 
+export type CertificateWithPoolOwnersAndRelays = {
+    certificate: CardanoTxCertificateType,
+    poolOwners: CardanoPoolOwnerProto[],
+    poolRelays: CardanoPoolRelayProto[],
+};
+
+export type PoolParametersWithOwnersAndRelays = {
+    poolParameters?: CardanoPoolParametersType,
+    poolOwners: CardanoPoolOwnerProto[],
+    poolRelays: CardanoPoolRelayProto[],
+};
+
 const transformPoolParameters = (
-    poolParameters: CardanoPoolParameters,
-): CardanoPoolParametersType => {
+    poolParameters?: CardanoPoolParameters,
+): PoolParametersWithOwnersAndRelays => {
+    if (!poolParameters) {
+        return { poolParameters: undefined, poolOwners: [], poolRelays: [] };
+    }
+
     validatePoolParameters(poolParameters);
 
     return {
-        pool_id: poolParameters.poolId,
-        vrf_key_hash: poolParameters.vrfKeyHash,
-        pledge: poolParameters.pledge,
-        cost: poolParameters.cost,
-        margin_numerator: poolParameters.margin.numerator,
-        margin_denominator: poolParameters.margin.denominator,
-        reward_account: poolParameters.rewardAccount,
-        owners: poolParameters.owners.map(owner => ({
+        poolParameters: {
+            pool_id: poolParameters.poolId,
+            vrf_key_hash: poolParameters.vrfKeyHash,
+            pledge: poolParameters.pledge,
+            cost: poolParameters.cost,
+            margin_numerator: poolParameters.margin.numerator,
+            margin_denominator: poolParameters.margin.denominator,
+            reward_account: poolParameters.rewardAccount,
+            owners: [],
+            relays: [],
+            metadata: poolParameters.metadata,
+            owners_count: poolParameters.owners.length,
+            relays_count: poolParameters.relays.length,
+        },
+        poolOwners: poolParameters.owners.map(owner => ({
```

Comment:
Since `poolOwners` and `poolRelays` are sent in different messages I needed to separate them.